packages/app-js/src/index.tsx

```diff
@@ -264,10 +287,49 @@ class App extends React.PureComponent<Props, State> {
       });
     };
   }
+
+  private generateLink = (): void => {
+    const { basePath } = this.props;
+    const {
+      props: { history, location },
+      state: { selected: { code } }
+    } = this;
+    const base64code = btoa(code);
+
+    if (base64code !== location.hash.substr(1)) {
+      history.push(`${basePath}#${base64code}`);
+    }
+    this.copyToClipboard(`${window.location.origin}/#${location.pathname}#${base64code}`);
+  }
+
+  private copyToClipboard = (link: string): void => {
```

Comment:
Can't we use the CopyToClipboard component here rather?

Reply:
I tried that... But I don't want to calculate the base64 value on every edit so that I can pass it as `value` for the `text` prop to the component.
That would be necessary because the CopyToClipboard component doesn't have an eventHandler so I can't calculate the base64 value of the code on click

Reply:
Grrr... yes, just looked it would have been great if the `text` param could have been string or function-returning-string

Reply:
Yup, it would ;). But that's how I ended up with the solution from Hackernoon. There really should be an easier native solution :|